src/ui/public/modals/confirm_modal.factory.js

```diff
@@ -0,0 +1,38 @@
+import { noop } from 'lodash';
+import uiModules from 'ui/modules';
+import template from './confirm_modal.html';
+import { Popover } from './popover';
+
+const module = uiModules.get('kibana');
+
+module.factory('confirmModal', function ($rootScope, $compile) {
+  let modalPopover;
+
+  return function confirmModal(message, onConfirm = noop, onCancel = noop, confirmButtonText = 'Confirm', cancelButtonText = 'Cancel') {
+
+    if (modalPopover) throw new Error('Ah ah ah, only one modal, buddy!');
```

Comment:
Can we make this error message clearer? E.g. "You've called confirmModal but there's already a modal open. You can only have one modal open at a time."